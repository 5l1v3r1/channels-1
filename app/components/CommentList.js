// @flow
import React from 'react';
import Comment from './core/Comment';
import CommentType from '../dataTypes/commentType';
import styles from './CommentList.scss';

const CommentList = ({
  comments
}: {
  comments: CommentType[]
}) => (
  <ul className={styles.commentList}>
    {comments.map(comment => (
      <Comment
        key={comment.id}
        author={comment.author}
        avatar={comment.avatar}
        description={comment.description}
        publishedAt={comment.publishedAt}
      />
    ))}
  </ul>
);

export default CommentList;
