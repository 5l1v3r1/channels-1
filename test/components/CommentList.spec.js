import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CommentList from '../../app/components/CommentList';

function setup() {
  const props = {
    comments: []
  };

  const enzymeWrapper = shallow(<CommentList {...props} />);

  return {
    props,
    enzymeWrapper
  };
}

describe('CommentList', () => {
  it('should render self with comments', () => {
    const { enzymeWrapper } = setup();
    enzymeWrapper.setProps({
      comments: [
        {
          id: 'z12ddf2arwvtvdk0s23iztxx5szwzppfj',
          author: 'Miron Swyst',
          avatar: 'https://yt3.ggpht.com/photo.jpg',
          description: 'background sound , ok , but a bit too interfering to ears* ...',
          publishedAt: 'a year ago'
        },
        {
          id: 'z12exdqh2tifxzexi04ch1kbaqm0sf25ao00k',
          author: 'gabo2212',
          avatar: 'https://yt3.ggpht.com/photo.jpg',
          description: 'Please no music on the background :(',
          publishedAt: 'a year ago'
        }
      ]
    });
    const tree = toJson(enzymeWrapper);

    expect(tree).toMatchSnapshot();
  });

  it('should render self with no comments', () => {
    const { enzymeWrapper } = setup();
    const tree = toJson(enzymeWrapper);

    expect(tree).toMatchSnapshot();
  });
});
