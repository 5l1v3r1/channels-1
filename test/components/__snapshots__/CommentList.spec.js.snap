// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentList should render self with comments 1`] = `
<ul
  className="commentList"
>
  <Comment
    author="Miron Swyst"
    avatar="https://yt3.ggpht.com/photo.jpg"
    description="background sound , ok , but a bit too interfering to ears* ..."
    publishedAt="a year ago"
  />
  <Comment
    author="gabo2212"
    avatar="https://yt3.ggpht.com/photo.jpg"
    description="Please no music on the background :("
    publishedAt="a year ago"
  />
</ul>
`;

exports[`CommentList should render self with no comments 1`] = `
<ul
  className="commentList"
/>
`;
